autometa/common/metabin.py

```diff
@@ -65,19 +67,38 @@ class MetaBin:
     root : str
         root name of `assembly` (Will remove common extension like '.fasta')
     nucls_fname : str
-        File name of `contigs` nucleotide ORFs
+        File name of contigs nucleotide ORFs
     prots_fname : str
-        File name of `contigs` amino-acid ORFs
+        File name of contigs amino-acid ORFs
     nucl_orfs_fpath : str
         </path/to/nucleotide/`outdir`/`nucls_fname`>
     prot_orfs_fpath : str
         </path/to/amino/acid/`outdir`/`prots_fname`>
     nseqs : int
         Number of contigs in MetaBin
 
+    Raises
+    ------
+    ValueError
+        One of `contig_ids` or `seqrecords` must be provided
+    ValueError
+        `contig_ids` do not match `seqrecords`
     """
 
-    def __init__(self, assembly, contigs, outdir=None):
+    def __init__(self, assembly, contig_ids=[], seqrecords=[], outdir=None):
+        if not contig_ids and not seqrecords:
+            raise ValueError("One of `contig_ids` or `seqrecords` must be provided!")
+        if contig_ids and seqrecords:
```

Comment:
I may discard the metabin changes. Previously I was using `MetaBin` to calculate the stats, but this is quite slow compared to the current implementation.

Reply:
The `if contig_ids and seqrecords` checks should only be performed if both are provided. Otherwise, one is retrieved from the other. using `else` would result in an error if only one of the two is provided.